onnxscript/optimizer/constant_folding_ir.py

```diff
@@ -0,0 +1,718 @@
+
+    if input is None or split is None or output is None:
+        return None
+
+    axis = _get_int_attribute(node, "axis", 0)
+    shape = input.shape
+    if shape is None:
+        return None
+    rank = len(shape)
+    if axis < 0:
+        axis = axis + rank
+    if axis < 0 or axis >= rank:
+        return None
+    split_dimension_size = shape[axis]
+    if not isinstance(split_dimension_size, int):
+        return None
+
+    split_value = _get_numpy_value(split)
+    if split_value is None:
+        return None
+    assert isinstance(split_value, np.ndarray)
+
+    if split_value.ndim == 0:
+        # split into chunks all of size 'split' if possible.
+        num_outputs = math.ceil(split_dimension_size / split_value.item())
+        split_outputs = [f"{output.name}_split_{i}" for i in range(num_outputs)]
+        split_values = op.Split(
+            input, axis=axis, num_outputs=num_outputs, outputs=split_outputs
+        )
+    elif split_value.ndim == 1:
+        # split into 'size(split)' chunks
+        num_outputs = split_value.size
+        split_outputs = [f"{output.name}_split_{i}" for i in range(num_outputs)]
+        split_values = op.Split(input, split, axis=axis, outputs=split_outputs)
+    else:
+        return None
+
+    keepdims = _get_int_attribute(node, "keepdims", 1)
+    if keepdims is None:
+        return None
+    if keepdims == 0:
+        # squeeze the split dimension if keepdims is 0
+        axis_val = op.Constant(value_int=axis, outputs=[f"{output.name}_axis"])
+        squeezed_values = []
+        for i in range(num_outputs):
+            squeezed = op.Squeeze(
+                split_values[i], axis_val, outputs=[f"{split_outputs[i]}_squeeze"]
+            )
+            squeezed_values.append(squeezed)
+        split_values = squeezed_values
+
+    logger.debug("SplitToSequence => Split + SequenceConstruct")
+
+    if isinstance(split_values, ir.Value):
+        split_values = [split_values]
+    return op.SequenceConstruct(*split_values)
+
+
+@register("SequenceAt")
+def sequence_at(op, node: ir.Node) -> ReturnValue:
+    input = node.inputs[0]
+    position = node.inputs[1]
+    output = node.outputs[0]
+    if input is not None and position is not None:
+        input_vals = input.symbolic_value
+        position_val = _get_numpy_value(position)
+        if isinstance(input_vals, list) and position_val is not None:
+            if position_val.size != 1:
+                return None
+            position_val = position_val.item()
+            try:
+                result = input_vals[position_val]
+            except IndexError:
+                return None
+            output.symbolic_value = result
+            logger.debug("SequenceAt %s => %s", input.name, result.name)
+            return op.Identity(result)
+    return None
+
+
+class ConstantFolder:
+    opset_imports: dict[str, int]
+
+    def __init__(
+        self,
+        external_data_folder: str,
+        do_shape_inference: bool,
```

Comment:
```suggestion
        shape_inference: bool,
```

nit: simplicity

Reply:
`shape_inference` by itself sounds ambiguous to me ... it is unclear what the option means. Maybe it is just me